### ISSUE TITLE
Class/Spec Secondary Resource Bars

### ISSUE THREAD

**Issue by MaDVitek** [Twintop/TwintopInsanityBar#297]
Can u add cat combo points for resto druid? And mb energy, but idk how u combine it with mana display

**Comment by Twintop** [comment]
Yes, having all of the Druid resources available to each spec depending on what form you're in (e.g. Energy+CP for Balance, with AP somewhere too (and maybe Mana as well?)) is something I'd like to add. For your example, you've hit the nail right on the head -- what happens to the Mana bar? Other specs have this issue in addition to Druids (e.g. Shadow Priests).

I've had in the backlog for a long time this issue #79 (GCD status bar) which, essenitally, is the same UI problem: where does this other bar go/live? There are a few possibilities I've been toying with:

1. Put it inside the main Bar layered on top of the primary resource and at the top or bottom.
2. Make them their own independent bars that can be placed above/below the primary bar (a la Combo Points) with their own borders and border colors.

Each of these would have separate options to toggle which extra bars are shown, of course.

My largest hesitation with the second option is the bloat of configuration options needed. I suppose I could always go with the first option and then expans it to the second option in the future.

Another consideration would be having threshold line configuration and bar text for each of the alternate forms. Just trying to use what the other specs have in place would be too problematic. Maybe another full config screen for the "shared" forms?

If you have any thoughts or suggestions, please don't hesitate to share.

**Comment by Twintop** [comment]
With the form weaving coming in TWW, this needs to be a higher priority.

**Comment by Twintop** [comment]
With #386 done, I *might* be able to make some version of this work sooner rather than later. @MaDVitek -- would the bar changing to reflect the current resource for the form you're in still be benefitial even without a secondary bar being displayed? This would be a step towards having a secondary bar, but I think hurdles have been removed on just switching the resource shown and having everything work correctly (threshold lines for abilities, resource display, maybe bar text?).

**Comment by Twintop** [comment]
@MaDVitek This will be available in Midnight.
